packages/realm_dart/lib/src/native/app_handle.dart

```diff
@@ -0,0 +1,592 @@
+    return customDataPtr.cast<Utf8>().toRealmDartString(freeRealmMemory: true)!;
+  }
+
+  Future<void> updateBaseUrl(Uri? baseUrl) {
+    final completer = Completer<void>();
+    using((arena) {
+      realmLib
+          .realm_app_update_base_url(
+            pointer,
+            baseUrl?.toString().toCharPtr(arena) ?? nullptr,
+            realmLib.addresses.realm_dart_void_completion_callback,
+            createAsyncCallbackUserdata(completer),
+            realmLib.addresses.realm_dart_userdata_async_free,
+          )
+          .raiseLastErrorIfFalse("Update base URL failed");
+    });
+    return completer.future;
+  }
+
+  Future<void> refreshCustomData(UserHandle user) {
+    final completer = Completer<void>();
+    realmLib
+        .realm_app_refresh_custom_data(
+          pointer,
+          user.pointer,
+          realmLib.addresses.realm_dart_void_completion_callback,
+          createAsyncCallbackUserdata(completer),
+          realmLib.addresses.realm_dart_userdata_async_free,
+        )
+        .raiseLastErrorIfFalse("Refresh custom data failed");
+    return completer.future;
+  }
+
+  String get id {
+    return realmLib.realm_app_get_app_id(pointer).cast<Utf8>().toRealmDartString()!;
+  }
+
+  Future<UserHandle> logIn(CredentialsHandle credentials) async {
+    final completer = Completer<UserHandle>();
+    realmLib
+        .realm_app_log_in_with_credentials(
+          pointer,
+          credentials.pointer,
+          realmLib.addresses.realm_dart_user_completion_callback,
+          createAsyncUserCallbackUserdata(completer),
+          realmLib.addresses.realm_dart_userdata_async_free,
+        )
+        .raiseLastErrorIfFalse("Login failed");
+    return await completer.future;
+  }
+
+  Future<void> registerUser(String email, String password) {
+    final completer = Completer<void>();
+    using((arena) {
+      realmLib
+          .realm_app_email_password_provider_client_register_email(
+            pointer,
+            email.toCharPtr(arena),
+            password.toRealmString(arena).ref,
+            realmLib.addresses.realm_dart_void_completion_callback,
+            createAsyncCallbackUserdata(completer),
+            realmLib.addresses.realm_dart_userdata_async_free,
+          )
+          .raiseLastErrorIfFalse();
+    });
+    return completer.future;
+  }
+
+  Future<void> confirmUser(String token, String tokenId) async {
+    final completer = Completer<void>();
+    using((arena) {
+      realmLib
+          .realm_app_email_password_provider_client_confirm_user(
+            pointer,
+            token.toCharPtr(arena),
+            tokenId.toCharPtr(arena),
+            realmLib.addresses.realm_dart_void_completion_callback,
+            createAsyncCallbackUserdata(completer),
+            realmLib.addresses.realm_dart_userdata_async_free,
+          )
+          .raiseLastErrorIfFalse();
+    });
+    return await completer.future;
```

Comment:
Yes. Debugging is slightly nicer, when we await before returning, but we return directly mos places.

Reply:
.. or actually we are all over the place on this one.

I think I would prefer we `await` before returning in `async` methods, but in this case the only reason the method is `async` is for this `await`, so I'll fix both.

Reply:
I just saw the methods around it don't await the future. I don't have strong opinions - it might be slightly more efficient to return the future directly rather than await it, but I don't imagine it's a major thing.

Reply:
I'll align